packages/@aws-cdk/aws-cloudtrail/lib/index.ts

```diff
@@ -178,7 +180,16 @@ export class Trail extends Resource {
       eventSelectors: this.eventSelectors
     });
 
-    this.trailArn = trail.trailArn;
+    const resourceIdentifiers = new ResourceIdentifiers(this, {
+      arn: trail.trailArn,
+      name: trail.trailName,
+      arnComponents: {
+        service: 'cloudtrail',
+        resource: 'trail',
+        resourceName: this.physicalName.value,
+      },
+    });
+    this.trailArn = resourceIdentifiers.arn;
```

Comment:
How do we enforce using this and not just `trail.trailArn`? Seems voluntary... I am worried that most people will not know to do this.

Reply:
That's fine; adding this later is a backwards-compatible change.